hugr-core/src/extension/prelude/generic.rs

```diff
@@ -0,0 +1,204 @@
+            let n_const = ConstUsize::new(n);
+            Some(vec![(0.into(), n_const.into())])
+        } else {
+            None
+        }
+    }
+}
+
+impl MakeOpDef for LoadNatDef {
+    fn from_def(op_def: &OpDef) -> Result<Self, OpLoadError>
+    where
+        Self: Sized,
+    {
+        crate::extension::simple_op::try_from_name(op_def.name(), op_def.extension_id())
+    }
+
+    fn init_signature(&self, _extension_ref: &Weak<Extension>) -> SignatureFunc {
+        let usize_t: Type = usize_custom_t(_extension_ref).into();
+        let params = vec![TypeParam::max_nat()];
+        PolyFuncTypeRV::new(params, FuncValueType::new(type_row![], vec![usize_t])).into()
+    }
+
+    fn extension_ref(&self) -> Weak<Extension> {
+        Arc::downgrade(&PRELUDE)
+    }
+
+    fn extension(&self) -> ExtensionId {
+        PRELUDE_ID
+    }
+
+    fn description(&self) -> String {
+        "Loads a generic bounded nat parameter into a usize runtime value.".into()
+    }
+
+    fn post_opdef(&self, def: &mut OpDef) {
+        def.set_constant_folder(*self);
+    }
+}
+
+/// Concrete load nat operation.
+#[derive(Clone, Debug, PartialEq)]
+pub struct LoadNat {
+    nat: TypeArg,
+}
+
+impl LoadNat {
+    fn new(nat: TypeArg) -> Self {
+        LoadNat { nat }
+    }
+}
+
+impl NamedOp for LoadNat {
+    fn name(&self) -> OpName {
+        LOAD_NAT_OP_ID
+    }
+}
+
+impl MakeExtensionOp for LoadNat {
+    fn from_extension_op(ext_op: &ExtensionOp) -> Result<Self, OpLoadError>
+    where
+        Self: Sized,
+    {
+        let def = LoadNatDef::from_def(ext_op.def())?;
+        def.instantiate(ext_op.args())
+    }
+
+    fn type_args(&self) -> Vec<TypeArg> {
+        vec![self.nat.clone()]
+    }
+}
+
+impl MakeRegisteredOp for LoadNat {
+    fn extension_id(&self) -> ExtensionId {
+        PRELUDE_ID
+    }
+
+    fn registry<'s, 'r: 's>(&'s self) -> &'r crate::extension::ExtensionRegistry {
+        &PRELUDE_REGISTRY
+    }
+}
+
+impl HasDef for LoadNat {
+    type Def = LoadNatDef;
+}
+
+impl HasConcrete for LoadNatDef {
+    type Concrete = LoadNat;
+
+    fn instantiate(&self, type_args: &[TypeArg]) -> Result<Self::Concrete, OpLoadError> {
+        match type_args {
+            [n] => Ok(LoadNat::new(n.clone())),
+            _ => Err(SignatureError::InvalidTypeArgs.into()),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        builder::{inout_sig, DFGBuilder, Dataflow, DataflowHugr},
+        extension::prelude::{usize_t, ConstUsize},
+        ops::{constant, OpType},
+        type_row,
+        types::TypeArg,
+        OutgoingPort,
+    };
+
+    use super::LoadNat;
+
+    #[test]
+    fn test_load_nat() {
+        let mut b = DFGBuilder::new(inout_sig(type_row![], vec![usize_t()])).unwrap();
+
+        let arg = TypeArg::BoundedNat { n: 4 };
+        let op = LoadNat::new(arg);
+
+        let out = b.add_dataflow_op(op, []).unwrap();
+
+        b.finish_hugr_with_outputs(out.outputs()).unwrap();
```

Comment:
Can we add some kind of assertion about the result, such as that is contains the expected operations?